cosmos/dbt/graph.py

```diff
@@ -378,6 +378,8 @@ def _add_vars_arg(self, cmd_args: list[str]) -> None:
         """
         if self.project.dbt_vars:
             cmd_args.extend(["--vars", json.dumps(self.project.dbt_vars, sort_keys=True)])
+        elif self.dbt_vars:
+            cmd_args.extend(["--vars", json.dumps(self.dbt_vars, sort_keys=True)])
```

Comment:
```suggestion
        if self.dbt_vars:
            cmd_args.extend(["--vars", json.dumps(self.dbt_vars, sort_keys=True)])
```
This should yield the same result, right?

@tatiana Was there a specific reason we checked for `dbt_vars` only in `self.project` instead of directly using `self.dbt_vars` while working on https://github.com/astronomer/astronomer-cosmos/pull/1114?

@AlexandrKhabarov While Tatiana provides insights on this, logically speaking, wouldn’t it make more sense to use the variables set at the project level during parsing/graph building rather than the ones set for the operator(I would incline to think that operator args would be used for each of the operator tasks during execution)? What do you think?

Reply:
@pankajkoti I believe it was a mistake to not just use `self.dbt_vars`, given the logic in https://github.com/astronomer/astronomer-cosmos/blob/56b386ec0aff009fc9b898df921e6cbb4a5551ac/cosmos/converter.py#L272

We probably should avoid using `self.project.dbt_vars` in this class and rely only on `self.dbt_vars